src/main/java/org/spongepowered/common/effect/particle/SpongeParticleEffect.java

```diff
@@ -58,98 +50,30 @@ public SpongeParticleType getType() {
         return this.type;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public Vector3d getMotion() {
-        return this.motion;
+    public <V> Optional<V> getOption(ParticleOption<V> option) {
+        return (Optional) Optional.ofNullable(this.options.get(option));
```

Comment:
@liach The `Map` has a `Object` as value type. So I had to cast it to the generic, though I could just have casted the object to the generic : `return Optional.ofNullable((V) this.options.get(option));`